odoo/addons/base/tests/test_db_cursor.py

```diff
@@ -33,21 +33,15 @@ def test_execute_bad_params(self):
 
 
 class TestTestCursor(common.TransactionCase):
-    @classmethod
-    def setUpClass(cls):
-        super(TestTestCursor, cls).setUpClass()
-        r = registry()
-        r.enter_test_mode(r.cursor())
-
-    @classmethod
-    def tearDownClass(cls):
-        r = registry()
-        r.test_cr.close()
-        r.leave_test_mode()
-        super(TestTestCursor, cls).tearDownClass()
-
     def setUp(self):
-        super(TestTestCursor, self).setUp()
+        super().setUp()
+        # make the registry in test mode
+        self.registry.enter_test_mode(self.cr)
+        self.addCleanup(self.registry.leave_test_mode)
+        # this is a test cursor
+        self.cr = self.registry.cursor()
+        self.addCleanup(self.cr.close)
+        self.env = odoo.api.Environment(self.cr, odoo.SUPERUSER_ID, {})
```

Comment:
Is there a particular reason the previous code could enter/leave the test_mode at the class level while we now do so on every test method ?

Reply:
Thanks for asking

Reply:
Before this, a new cursor was created in `setUp`, and because the registry was in test mode, it was a test cursor.
Now the test's cursor is created in `setUpClass`, where the registry is not in test mode yet, and I wanted to allocate a new test cursor for each test.